lambdas/access_counts/index.py

```diff
@@ -0,0 +1,200 @@
+        name,
+        CAST(histogram(date) AS JSON) AS counts
+    FROM object_access_log JOIN package_hashes
+    ON object_access_log.bucket = package_hashes.bucket AND key = '.quilt/packages/' || hash
+    GROUP BY eventname, package_hashes.bucket, name
+""")
+
+PACKAGE_VERSION_ACCESS_COUNTS = textwrap.dedent("""\
+    SELECT
+        eventname,
+        package_hashes.bucket AS bucket,
+        name,
+        version,
+        hash,
+        CAST(histogram(date) AS JSON) AS counts
+    FROM object_access_log JOIN package_hashes
+    ON object_access_log.bucket = package_hashes.bucket AND key = '.quilt/packages/' || hash
+    GROUP BY eventname, package_hashes.bucket, name, version, hash
+""")
+
+
+athena = boto3.client('athena')
+s3 = boto3.client('s3')
+
+
+def run_query(query_string):
+    output = 's3://%s/%s/' % (QUERY_RESULT_BUCKET, QUERY_TEMP_DIR)
+
+    response = athena.start_query_execution(
+        QueryString=query_string,
+        QueryExecutionContext=dict(Database=ATHENA_DATABASE),
+        ResultConfiguration=dict(OutputLocation=output)
+    )
+    print("Started query:", response)
+
+    execution_id = response['QueryExecutionId']
+
+    return execution_id
+
+
+def wait_for_query(execution_id):
+    while True:
+        response = athena.get_query_execution(QueryExecutionId=execution_id)
+        print("Query status:", response)
+        state = response['QueryExecution']['Status']['State']
+
+        if state == 'RUNNING':
+            pass
+        elif state == 'SUCCEEDED':
+            break
+        elif state == 'FAILED':
+            raise Exception("Query failed! QueryExecutionId=%r" % execution_id)
+        elif state == 'CANCELLED':
+            raise Exception("Query cancelled! QueryExecutionId=%r" % execution_id)
+        else:
+            assert False, "Unexpected state: %s" % state
+
+        time.sleep(5)
+
+
+def delete_temp_dir():
+    params = dict(
+        Bucket=QUERY_RESULT_BUCKET,
+        Prefix=QUERY_TEMP_DIR,
+        MaxKeys=1000,  # The max we're allowed to delete at once.
+    )
+    while True:
+        list_response = s3.list_objects_v2(**params)
+        contents = list_response.get('Contents')
+        if not contents:
+            break
+
+        delete_response = s3.delete_objects(
+            Bucket=QUERY_RESULT_BUCKET,
+            Delete=dict(
+                Objects=[dict(
+                    Key=obj['Key']
+                ) for obj in contents]
+            )
+        )
+        errors = delete_response.get('Errors')
+        if errors:
+            print(errors)
+            raise Exception("Failed to delete the temporary directory")
+
+        if list_response['IsTruncated']:
+            params.update(dict(
+                ContinuationToken=list_response['ContinuationToken']
+            ))
+        else:
+            break
+
+
+def handler(event, context):
+    delete_temp_dir()
+
+    for query_id in [run_query(DROP_OBJECT_ACCESS_LOG), run_query(DROP_PACKAGE_HASHES)]:
```

Comment:
worth noting that this is safe since it happens in an athena db owned by the stack? (i.e. `object_access_logs` could be a common table name).

Reply:
Ok